### ISSUE TITLE
[Improve] Graceful stop flink job with savepoint

### ISSUE THREAD

**Issue by dyccode** [apache/streampark#3596]
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

The stop interface of Flink can elegantly stop jobs.The biggest difference between the cancel interface and the stop interface is that the stop interface is used to stop the task. After the savepoint is generated, the job can wait for `notifyCheckpointComplete()` to complete the execution.So, I want to use the stop interface by default when using savepoint to stop Flink jobs.

### Usage Scenario

When using savepoint to stop Flink jobs.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by dyccode** [comment]
Does the`FINISHED` state mean that the job has been permanently stopped?

**Comment by wolfboys** [comment]
Do you want to improve this issue? Looking forward to your PR! 😊

**Comment by wolfboys** [comment]
> Does the`FINISHED` state mean that the job has been permanently stopped?

Yes, I think so, The status indicates that the flink job has been thoroughly completed and stopped. 👍

**Comment by dyccode** [comment]
>Yes, I think so, The status indicates that the flink job has been thoroughly completed and stopped. 👍

Is the `FINISHED` status consistent with the status of Flink? When Flink uses the stop interface to stop a job, the final state of the task is `FINISHED`. Should the state in streampark be consistent with Flink's state?

**Comment by dyccode** [comment]
@wolfboys Due to being quite busy recently, I only started working on this issue yesterday.When I finished the development and started testing my code, I found that the program wouldn't start. I discovered that the startup configuration file's name had been changed, but the program was still trying to read the previous name. I am using the latest code.Are there any specific considerations or precautions I should take when using the latest code? I look forward to your response.
